pkg/clusteragent/admission/mutate/common_test.go

```diff
@@ -152,6 +153,137 @@ func Test_injectEnv(t *testing.T) {
 	}
 }
 
+func Test_shouldInject(t *testing.T) {
+	var mockConfig *config.MockConfig
+	tests := []struct {
+		name        string
+		pod         *corev1.Pod
+		setupConfig func()
+		want        bool
+	}{
+		{
+			name:        "mutate unlabelled, no label",
+			pod:         fakePodWithLabel("", ""),
+			setupConfig: func() { mockConfig.Set("admission_controller.mutate_unlabelled", true) },
+			want:        true,
```

Comment:
shouldn't this be false for injecting the APM libraries according to the RFC?

Reply:
No, we are not changing existing logic around using mutateUnlabelled and pod label - they still can be used for local library injection.

Reply:
the APM library case is handled separately in #19595 so this isn't an issue